### PR TITLE
Adding first models for zendesk and zoho

### PR DESCRIPTION
**Pull request by gui-brandalley**

### CODE REVIEW

File: dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	materialized='incremental',
+	unique_key='id',
+	partition_by = {
+      "field": "created_at",
+      "data_type": "timestamp",
+      "granularity": "day"
+    }
+) }}
+
+{% set min_ts = '2023-02-01' %}
```
Comment (nicklupu) [inline dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql:12]: ill ignore checking incremental logic for now until its actually implemented!

File: dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		where updated_at >= ( select IF(max(gravity_inserted)>max(gravity_updated), max(gravity_inserted), max(gravity_updated)) from {{this}} )
+	)
+  {% endset %}
+  {% set result = run_query(sql) %}
+  {% set min_ts = result.columns['min_ts'][0]  %}
+{% endif %}
+*/
+
+with tickets as (
+    select  Id,
```
Comment (nicklupu) [inline dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql:32]: do you think we could start renaming columns as snake_case when we're ingesting new models? its a mixture of both atm i know but i think majority of places its snake_case so think something will ultimately want to make consistent going forwards

File: dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		where updated_at >= ( select max(updated_at) from {{this}} )
+	)
+  {% endset %}
+  {% set result = run_query(sql) %}
+  {% set min_ts = result.columns['min_ts'][0]  %}
+{% endif %}
+*/
+
+with tickets as (
+    select  id,
```
Comment (nicklupu) [inline dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql:32]: i think we should start adding unique tests (and not null) on all the primary keys in model configs too as quite a few models do have dupes currently that wouldve been caught by these tests. BQ have that you can set the primary key in a table directly now, so may start to add these into the tables themselves and then wont need to test it for primary keys

File: dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql (hunk 2 above)
Comment (nicklupu) [reply dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql:32]: maybe lets book in some time to plan out what we want as best practices for our models and i can chuck it all in the readme after?

File: dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql (hunk 3 above)
Comment (gui-brandalley) [reply dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql:32]: Ok, so this is something we can do in BQ directly or should be done in dbt too?

File: dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql (hunk 1 above)
Comment (gui-brandalley) [reply dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql:12]: Yes, will update straight after the first run.

File: dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql (hunk 3 above)
Comment (nicklupu) [reply dbt/brandalley-dbt/models/zendesk/zendesk_tickets_detailed.sql:32]: do it in dbt for now and then might deprecate it in the near future - need to do a bit more research on it first tho and work out if we want to use it/how

File: dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql (hunk 2 above)
Comment (nicklupu) [reply dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql]: ok cool, well i can always do a draft of what think and then we can discuss it after and see if you agree/have any other thoughts?

File: dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+/* On hold for first run
+{% set min_ts = '2023-02-01' %}
+{% if execute and is_incremental() %}
+  {% set sql %}
+    -- Query to see the earliest event date that needs to be rebuilt from for inserted tickets since last run  
+    select min(updated_at) as min_ts from (
+		select 
+			IF(min(gravity_inserted)<min(gravity_updated), min(gravity_inserted), min(gravity_updated)) as updated_at
+		from     {{ source(
```
Comment (nicklupu) [inline dbt/brandalley-dbt/models/zohodesk/zohodesk_tickets_detailed.sql:19]: i think it might still be trying to compile this source even tho its commented out and thats why its failing

File: dbt/brandalley-dbt/models/zendesk/_docs.yml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        description: "Ids of tickets merged"
+      - name: status
+        description: "Status of the ticket"
+      - name: subject
+        description: "Subject of the ticket"
+      - name: updated_at
+        description: "Update date"
+        meta:
+          dimension:
+            type: date
```
[the remark this answers is not in the document]
Comment (gui-brandalley) [reply dbt/brandalley-dbt/models/zendesk/_docs.yml:49]: Date is working also with hours in LD. I remember thinking the same in the past.

**Review by nicklupu — APPROVED** [review]
